httpclient/src/main/java/com/okta/sdk/impl/http/httpclient/HttpClientRequestExecutor.java

```diff
@@ -429,21 +429,32 @@ private URI getRedirectUri(HttpResponse httpResponse) {
     private void pauseBeforeRetry(int retries, HttpResponse httpResponse, long timeElapsed) throws RestException {
         long delay = -1;
         long timeElapsedLeft = maxElapsedMillis - timeElapsed;
+
+        // check before continuing
+        if (!shouldRetry(retries, timeElapsed)) {
+            throw failedToRetry();
+        }
+
         if (backoffStrategy != null) {
             delay = Math.min(this.backoffStrategy.getDelayMillis(retries), timeElapsedLeft);
         } else if (httpResponse != null && httpResponse.getStatusLine().getStatusCode() == 429) {
             delay = get429DelayMillis(httpResponse);
             if (!shouldRetry(retries, timeElapsed + delay)) {
-                throw new RestException("Cannot retry request, next request will exceed retry configuration.");
+                throw failedToRetry();
             }
             log.debug("429 detected, will retry in {}ms, attempt number: {}", delay, retries);
         }
 
-        // default / fallback strategy
+        // default / fallback strategy (backwards compatible implementation)
         if (delay < 0) {
             delay = Math.min(getDefaultDelayMillis(retries), timeElapsedLeft);
         }
 
+        // this shouldn't happen, but guard against a negative delay at this point
+        if (delay < 0) {
+            throw failedToRetry();
```

Comment:
I'm assuming this means the call fails?

Reply:
correct the `failedToRetry()` method just returns a new exception